doorstop/core/item.py

```diff
@@ -163,6 +164,14 @@ def _set_attributes(self, attributes):
                 value = Text(value)
             elif key == 'ref':
                 value = value.strip()
+            elif key == 'references':
+                if not isinstance(value, list):
```

Comment:
I would not silently accept this. I would throw an exception.

Reply:
I expected this kind of a comment to ask a more general question: what is the best place to validate the `references` parameter?

In addition to checking that the input is a list, more things could be checked, such as:

- each element of the `references` list shall be a dictionary
- `type` has to be of a `file` value (currently the one supported)
- `path` shall be a non-empty string

Do we want to include all of these here this MR or a simple exception like @sebhub is suggesting is enough?

Reply:
I think the program should reject all unexpected input. If you silently ignore this (e.g. because a future version of Doorstop added new things which you don't understand), then this may cause unexpected effects which are hard to trace for a user.

I would iterate through the list and empty the dictionaries with pop(). If exceptions occur, then print a message and terminate.